OpenTrons/opentrons-api
# refactor(protocol-engine): Implement `magneticModule/disengageMagnet`

# Overview

This implements the `magneticModule/disengageMagnet` Protocol Engine command, whose skeleton was added in #9652. This closes part of #8243.

# Changelog

* Implement `magneticModule/disengageMagnet`.
* Refactor `ModuleView` and `magneticModule/engageMagnet` to simplify things that are common to both Magnetic Module commands: validating that the given ID points to a module, validating that the ID points to a *magnetic* module, and handling the case of virtualized modules.

# Manual test plan

1. Load a Magnetic Module or two with `loadModule` commands.
2. Issue `magneticModule/disengageMagnet` commands and confirm that they home the modules properly.
3. Try interleaving them with `magneticModule/engageMagnet` and confirm that the height tracking works properly.
4. Check to see that the command gives a helpful error when you supply a bogus `moduleId`.
5. Check to see that the command gives a helpful error when you supply a `moduleId` that points to a non-Magnetic module.
6. Check to see that the command gives a helpful error when you physically disconnect the module before running it (but after you've successfully loaded the module).

This can be combined with the manual test plan in #9599, which has copy-pastable request bodies.

# Risk assessment

Low. No production code uses this yet.
